### PR TITLE
Add option to set indent depth of closing brackets

### LINKED ISSUE #501: Option to indent the closing bracket of a multiline element by +1
## Request / Idea
Thanks for the plugin. We are migrating a sizeable codebase to use it. One currently impossible style opinion would be really helpful for us: indenting the trailing `)` of a multiline element to the same indent level as the props.

## Input

```pug
ExampleComponent(
  variant="danger"
  title="Oh no, a connection error!"
)
```

## Expected Output

```pug
ExampleComponent(
  variant="danger"
  title="Oh no, a connection error!"
  )
```

## Crude Implementation

```diff
// Line 1171 in printer.ts
- this.result += this.indentString.repeat(this.indentLevel);
+ this.result += this.indentString.repeat(this.indentLevel + 1);
```

## Additional Context

We feel this is preferable over the current default indent position as it allows you to both comment out the whole component one one go...

```pug
// Errors
// ExampleComponent(
  variant="danger"
  title="Oh no, a connection error!"
)
```

```pug
// Works
// ExampleComponent(
  variant="danger"
  title="Oh no, a connection error!"
  )
```

...as well as retains the ability to re-order the props easily without having to manually move the closing bracket to the end of the last prop.

Could this style be supported as an option? The code is trivial, so it's more of a bike-shedding question.

### PR DESCRIPTION
**Pull request by MrHaila**
Closes #501 

Adds a new option to modify how deep the closing brackets are indented when `bracketSameLine` is not used. Defaults (= null) to the current behavior and allows the choices of `0` (current default) or `1` (indent by +1).

I chose to add the feature as a numeric depth parameter instead of a true/false flag because it's a good way to communicate the feature, and there seem to be issues with passing booleans into Prettier plugins. Added tests to verify the new option.

Feedback is very much welcome if this approach makes sense!

### CODE REVIEW

File: tests/options/pugClosingBracketIndentDepth/pug-closing-bracket-indent-depth.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { compareFiles } from 'tests/common';
+import { describe, expect, it } from 'vitest';
+
+describe('Options', () => {
+  describe('bracketSameLine', () => {
```
Comment (Shinigami92) [inline tests/options/pugClosingBracketIndentDepth/pug-closing-bracket-indent-depth.test.ts:5]: **suggestion:** 
```suggestion
  describe('pugClosingBracketIndentDepth', () => {
```

File: tests/options/pugClosingBracketIndentDepth/pug-closing-bracket-indent-depth.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    it('should handle pugBracketSameLine:0', async () => {
```
Comment (Shinigami92) [inline tests/options/pugClosingBracketIndentDepth/pug-closing-bracket-indent-depth.test.ts:6]: **suggestion:** 
```suggestion
    it('should handle pugClosingBracketIndentDepth:0', async () => {
```

and below

**Comment by MrHaila** [comment]
Test names fixed 🙏

**Comment by MrHaila** [comment]
Looks like I mistakenly didn't commit in an unformatted input file for the tests, thus failing them for CI. Fixed. 🤦‍♂️

**Comment by Shinigami92** [comment]
I'll have a last in-depth look tomorrow

**Comment by MrHaila** [comment]
Thanks for the cleanup commit. It makes sense. Unfortunately I literally can't see the UI for enabling upstream commits as per Github's documentation.

Maybe our org has some global setting to disable it. Should have forked on my personal account instead. Sorry about that 😅

**Comment by Shinigami92** [comment]
> Thanks for the cleanup commit. It makes sense. Unfortunately I literally can't see the UI for enabling upstream commits as per Github's documentation.
> 
> Maybe our org has some global setting to disable it. Should have forked on my personal account instead. Sorry about that 😅

Yeah that's it, it needs to be from a non-org account 👌

**Review by Shinigami92 — APPROVED** [review]

**Comment by Shinigami92** [comment]
Merging this now, but I need to cleanup some more things before the next release. So potentially a release could come around weekend, depending on how much time I can spend.

**Pull request merged**